src/core/lib/iomgr/error.cc

```diff
@@ -70,12 +70,42 @@ absl::Status grpc_os_error(const grpc_core::DebugLocation& location, int err,
 }
 
 #ifdef GPR_WINDOWS
+std::string WSAErrorToShortDescription(int err) {
```

Comment:
Does Windows not provide a function that does this already?  I would have expected that it would.  Posix has had `strerror()` for a long time.

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage

Reply:
We use `FormatMessage` in `gpr_format_message`, but it produces longer messages. For example, for the `WSAECONNRESET` error, FormatMessage produces "An existing connection was forcibly closed by the remote host", but for a top-level Status summary (which becomes the error strings that clients receive), something like "Connection reset" will do for a summary. The summary seems friendlier.

Here is an example error:

```
I0620 18:33:23.213000000 15608 src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1609] 000001946F442850 CANCEL PINGS: UNKNOWN:Delayed close due to in-progress write {created_time:"2023-06-21T01:33:23.211759741+00:00", file_line:662, file:"src/core/ext/transport/chttp2/transport/chttp2_transport.cc", children:[UNKNOWN:Endpoint read failed {file:"src/core/ext/transport/chttp2/transport/chttp2_transport.cc", file_line:2534, created_time:"2023-06-21T01:33:23.210383893+00:00", occurred_during_write:0, children:[UNAVAILABLE:Connection Reset {file:"src/core/lib/iomgr/tcp_windows.cc", file_line:189, created_time:"2023-06-21T01:33:23.2102455+00:00", wsa_error:10054, grpc_status:14, os_error:"An existing connection was forcibly closed by the remote host.\r\n", syscall:"IOCP/Socket"}]}, UNAVAILABLE:Connection Reset {file:"src/core/lib/iomgr/tcp_windows.cc", file_line:425, created_time:"2023-06-21T01:33:23.212417741+00:00", wsa_error:10054, grpc_status:14, os_error:"An existing connection was forcibly closed by the remote host.\r\n", syscall:"WSASend"}]}    
```

This propagates to the surface as an error with status `GRPC_STATUS_UNAVAILABLE` and message "Connection Reset".